pkg/cloud/vsphere/services/userdata/controlplane.go

```diff
@@ -292,6 +342,17 @@ func JoinControlPlane(input *ContolPlaneJoinInput) (string, error) {
 	return userData, err
 }
 
+func NewCloudConfig(input *CloudConfigInput) (string, error) {
```

Comment:
I'm curious why this is its own function as opposed to part of the cloud init user data func. Just to make testing easier?

Reply:
eyah

Reply:
I saw that 'eyah'. I thought you were pulling a Howard Dean and excitedly answering :)

Reply:
haha, btw I forgot to add comments for this exported func, just pushed a change